src/app/clusters/network-commissioning/network-commissioning.cpp

```diff
@@ -1150,6 +1152,60 @@ void Instance::HandleQueryIdentity(HandlerContext & ctx, const Commands::QueryId
 }
 #endif // CHIP_DEVICE_CONFIG_ENABLE_WIFI_PDC
 
+Instance::InstanceRecord * Instance::sInstanceRecord = nullptr;
+
+void Instance::RecordNetwork(Instance * currentNetwork)
+{
+    InstanceRecord * curr = sInstanceRecord;
+    while (curr)
+    {
+        if (curr->mInstance == currentNetwork)
+        {
+            return;
+        }
+        curr = curr->next;
+    }
+
+    InstanceRecord * newInstanceRecord = reinterpret_cast<InstanceRecord *>(Platform::MemoryAlloc(sizeof(InstanceRecord)));
```

Comment:
Why do we need the heap alloc?  Why not just have Instance have an mNext pointer?

Reply:
Use mNext pointer instead.